indexer/Indexer.cc

```diff
@@ -413,6 +405,49 @@ void TuIndexer::saveNestedNameSpecifier(
   }
 }
 
+void TuIndexer::saveRecordDecl(const clang::RecordDecl &recordDecl) {
+  this->saveTagDecl(recordDecl);
+  // Superclass declarations will be visited during Visit*TypeLoc methods.
+}
+
+void TuIndexer::saveRecordTypeLoc(const clang::RecordTypeLoc &recordTypeLoc) {
+  this->saveTagTypeLoc(recordTypeLoc);
+}
+
+void TuIndexer::saveTagDecl(const clang::TagDecl &tagDecl) {
+  auto optSymbol = this->symbolFormatter.getTagSymbol(tagDecl);
+  if (!optSymbol.has_value()) {
+    return;
+  }
+  auto symbol = optSymbol.value();
+
+  if (!tagDecl.isThisDeclarationADefinition()) {
+    // Forward declarations should be marked as references. In the future,
+    // we should add an extra role to SCIP for forward declarations.
```

Comment:
Would be great to make this a todo/fixme-what-when. So something like:

```
// TODO: Mark these as forward declarations instead of
// references when SCIP has a forward declaration role.
// https://github.com/sourcegraph/scip/issues/nnnn
```

That way it is easy for someone to assess when fixing this is unblocked, how to push progress along if they care, etc.

Reply:
Added link to GitHub issue.